wrappers/rust/icicle-cuda-runtime/src/memory.rs

```diff
@@ -205,9 +227,8 @@ impl<T, const D_ID: usize> DeviceSlice<T, D_ID> {
     }
 }
 
-impl<T, const D_ID: usize> DeviceVec<T, D_ID> {
+impl<T> DeviceVec<T> {
     pub fn cuda_malloc(count: usize) -> CudaResult<Self> {
```

Comment:
Should this and `cuda_malloc_async` be private now that we have `cuda_malloc_for_device<_async>`?

Reply:
I don't know... If you're only using one card, there's no need in the `for_device` version of these methods. I personally would use `cuda_malloc_async` in multi-device settings too.